src/libraries/System.Private.Xml/src/System/Xml/Core/XmlReaderAsync.cs

```diff
@@ -371,46 +365,46 @@ internal async Task<string> InternalReadContentAsStringAsync()
                         // skip comments, pis and end entity nodes
                         break;
                     case XmlNodeType.EntityReference:
-                        if (this.CanResolveEntity)
+                        if (CanResolveEntity)
                         {
-                            this.ResolveEntity();
+                            ResolveEntity();
                             break;
                         }
                         goto default;
-                    case XmlNodeType.EndElement:
                     default:
                         goto ReturnContent;
                 }
-            } while ((this.AttributeCount != 0) ? this.ReadAttributeValue() : await this.ReadAsync().ConfigureAwait(false));
+            } while (AttributeCount != 0 ? ReadAttributeValue() : await ReadAsync().ConfigureAwait(false));
 
-        ReturnContent:
-            return (sb == null) ? value : sb.ToString();
+            ReturnContent:
```

Comment:
This change goes against the coding style: https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/coding-style.md

When using labels (for goto), indent the label one less than the current indentation.